### DISCUSSION TITLE
Allow specifying multiple power_in pins for different itemcomponents

### DISCUSSION THREAD

**Discussion by G-M-twostay** [FakeFishGames/Barotrauma#14103] [Code questions and modding support]
Can we add an option to specify which power pin each component draws power from? Currently, all components are hard-coded to draw power from "power_in" pin. This can be problematic when creating subs/mods that need intricate power systems. For example, currently, if I make gun lights need power, then they would draw power from the same pin with the turret itself. This means that they'll draw power from the supercapacitors, or I have to connect it to some power source, but this makes the turret consume most power from the secondary power source, rendering the supercapacitor quite useless.

I have identified what parts of the code need changes. https://github.com/FakeFishGames/Barotrauma/blob/4a63dacbcee8e3829a44e9c16a5335704a979a4d/Barotrauma/BarotraumaShared/SharedSource/Items/Components/Power/Powered.cs#L226 is where the power pin is hard coded to be "power_in". I suggest adding a new attribute to `Powered` component that allows you to specify which pin to choose as the power connection.

**Comment by SomeRandomNoobKekeke** [comment]
If you just want turret lights to turn off without power then you can just give it 1-2kw consumption, it's almost not noticeable on capacitor

**Reply by G-M-twostay** [reply]
your approach kind of works, but separating 2 power pins is still better.
